cmake/ccache.cmake

```diff
@@ -8,45 +9,62 @@ if (CMAKE_CXX_COMPILER_LAUNCHER MATCHES "ccache" OR CMAKE_C_COMPILER_LAUNCHER MA
     return()
 endif()
 
-option(ENABLE_CCACHE "Speedup re-compilations using ccache (external tool)" ON)
+option(ENABLE_CCACHE "Deprecated, use COMPILER_CACHE=(auto|ccache|sccache|disabled)")
```

Comment:
Maybe I don't see it, but where is `ENABLE_CCACHE` evaluated (besides l. 13-15)?

Reply:
`OFF` by default for options https://cmake.org/cmake/help/latest/command/option.html

Reply:
What I meant is that `ENABLE_CCACHE` is not evaluated at all in this file. Whatever the user sets or doesn't set, it won't have an effect. This is not "deprecating" an option, it is abolishing it.

Reply:
It's bool and can have either 1 or 0. How could we deprecate it without setting it to 0 and looking for 1?

If it's 1|ON, it's equal to default `COMPILER_CACHE=auto`. If it's unset, well, one will replace it in their script